metadata-ingestion/src/datahub/ingestion/source/tableau/tableau_common.py

```diff
@@ -1000,3 +1001,19 @@ def get_filter_pages(query_filter: dict, page_size: int) -> List[dict]:
         ]
 
     return filter_pages
+
+
+def optimize_query_filter(query_filter: dict) -> dict:
+    """
+    Duplicates in the filter cause duplicates in the result,
+    leading to entities/aspects being emitted multiple times unnecessarily
+    """
+    optimized_query = copy.deepcopy(query_filter)
+
+    if query_filter.get(c.ID_WITH_IN):
+        optimized_query[c.ID_WITH_IN] = list(set(query_filter[c.ID_WITH_IN]))
```

Comment:
could we use our OrderedSet type here? mainly to ensure consistent behavior across runs and make testing/mocking easier

Reply:
done in commit 877a50a